File: packages/service/src/domain/commands/model/command.ts
class Command {
  key: string;
  rawArgs: (string | null);

  constructor(key: string, rawArgs: (string | null)) {
    this.key = key;
    this.rawArgs = rawArgs;
  }

  getArgsByDelimiter(delimiter: string): string[] {
    return (this.rawArgs || '')
      .split(delimiter)
      .map((s) => s.trim())
      .filter((s) => (s.length > 0));
  }
}

export { Command };

File: service/src/main/kotlin/xyz/lebkuchenfm/domain/commands/model/Command.kt
package xyz.lebkuchenfm.domain.commands.model

data class Command(
    val key: String,
    val rawArgs: String?,
) {
    fun getArgsByDelimiter(delimiter: String): List<String> {
        return rawArgs
            ?.split(delimiter)
            ?.map { it.trim() }
            ?.filter { it.isNotEmpty() }
            ?: emptyList()
    }
}
